### PR TITLE
Elaborate on lock period lengths in voluntary locking section

### LINKED ISSUE #3041: Explain how long a lock period is in voluntary locking / conviction voting section
Here: 
https://wiki.polkadot.network/docs/learn-governance#voluntary-locking

It just sys how many lock periods ( = launch periods). We should either grab the value directly from the same variable as the parameters page gets it or link the parameters page ( https://wiki.polkadot.network/docs/kusama-parameters or https://wiki.polkadot.network/docs/maintain-polkadot-parameters)

### PR DESCRIPTION
**Pull request by alfarok**
### Task
This PR addresses #3041.

### Goals

- [x] The current [voluntary-locking](https://wiki.polkadot.network/docs/learn-governance#voluntary-locking) section provides details regarding how long a lock period will persist.
- [x] Lock period length is sourced directly from the [polkadot parameters page](https://wiki.polkadot.network/docs/maintain-polkadot-parameters) and [kusama parameters page](https://guide.kusama.network/docs/kusama-parameters/)or links to it

### CODE REVIEW

File: docs/general/grants.md [hunk 1]
```diff
@@ -45,7 +45,6 @@ If you are interested in obtaining KSM for building or research, you can apply t
 
 Below is a list of other grant programs in the Polkadot/Substrate ecosystem:
 
-- [Darwinia Grants Program](https://docs.darwinia.network/developers/dev-bounty#grant-program)
```
Comment (alfarok) [inline docs/general/grants.md]: It appears like they have removed this page

File: docs/maintain/kusama/maintain-guides-how-to-validate-kusama.md [hunk 2]
```diff
@@ -589,8 +589,7 @@ increase the number of KSM staked or seek out nominators for your validator in o
 validator set.
 
 **Congratulations**, if you have followed all of these steps, and been selected to be a part of the
-validator set, you are now running a Kusama validator! If you need help, reach out on the
-[Kusama forum](https://forum.kusama.network/) or in the
```
Comment (alfarok) [inline docs/maintain/kusama/maintain-guides-how-to-validate-kusama.md]: forum no longer relevant

**Comment by DrW3RK** [comment]
Good work with catching broken links! Just merged another PR that fixed some of them.

I have some feedback regarding the governance page update. When a reader lands on the voluntary locking section, the information presented there should be intuitive. Right now, the reader has to first look at the table that has locked periods and then read further into the document to access the links where the number of days in a lock period are listed.

To improve the experience, we can add an extra column here with the days displayed through conditional rendering for Polkadot and Kusama.

image

If you like to push further and explore more, you could also fetch the voteLockingPeriod from the chain state and use it to calculate the values rendered in the table. 115,200 x 6 seconds (block time of Kusama) = 8 days. 

image

Also, limit a PR to address a single issue and avoid mixing it up with unrelated issues. We can discuss further during the team sync. Talk to you soon!

File: docs/learn/learn-governance.md [hunk 3]
```diff
@@ -258,8 +258,9 @@ double.
 
 The maximum number of "doublings" of the lock period is set to 6 (and thus 32 lock periods in
 total), and one lock period equals {{ polkadot: 28 days :polkadot }}{{ kusama: 8 days :kusama }}. 
-Only doublings are allowed; you cannot lock for, say, 24 periods and increase your conviction 
-by 5.5, for instance.
+Only doublings are allowed; you cannot lock for, say, 24 periods and increase your conviction by
+5.5. For additional information on lock period lengths check out the governance table on the
+{{ polkadot: [Polkadot Parameters page](../maintain-polkadot-parameters/#governance) :polkadot }}{{ kusama: [Kusama Parameters page](../kusama-parameters/#governance) :kusama }}.
```
Comment (profradha) [inline docs/learn/learn-governance.md]: Lock period length is not referenced anywhere in the parameters page

**Review by DrW3RK — CHANGES_REQUESTED** [review]
Added a comment earlier. You can create a separate PR for modifying governance page and limit this PR to broken link fixes

**Comment by DrW3RK** [comment]
Forgot to link the PR that was already merged that was addressing broken links on Substrate docs https://github.com/w3f/polkadot-wiki/pull/3437

**Comment by alfarok** [comment]
> I have some feedback regarding the governance page update. When a reader lands on the voluntary locking section, the information presented there should be intuitive. Right now, the reader has to first look at the table that has locked periods and then read further into the document to access the links where the number of days in a lock period are listed.
> 

Thanks for the feedback here!  I wasn't sure exactly based on the original issue how much detail was required or how common it was to duplicate the information vs linking back and forth.  I will work on getting this polished.

**Comment by alfarok** [comment]
@DrW3RK Also thank you for pointing me at #3437 I wasn't aware of this.  ~~I think it is still missing updates from several old substrate docs but I will pull in those changes now that it's merged to compare and report back.~~  Looks like all the remaining `doc.substrate.io` references are correct.  Combined with the parity docs replacements, the test is no longer showing redirects from that domain.

**Comment by DrW3RK** [comment]
Sure! I personally think some level of redundancy helps in the sections of the Wiki to minimize the number of links the reader has to click through. So, feel free to add redundant content in the context of the particular sections in the Wiki and then link it to the exclusive section for further reading.

I always try to read the PR changes from the user perspective and when I clicked through the parameters link to find information about voting lock period and I could not find any. I'd encourage you to wear the shoes of a general reader who would be using the Wiki when adding the content. Talk to you tomorrow!

**Comment by alfarok** [comment]
@DrW3RK I moved the other work into a separate PR #3442 for clarity.  Originally when I thought this task just required referencing the parameters pages I thought it made sense to do all the link restoration together but I totally agree separate PR's is the way the go.

One more question related to this work.  Can you point me to an existing example where dynamic values are rendering in markdown?  I would like to try fetch the `voteLockingPeriod` from the chain state but didn't know this was possible in markdown.  Is it a feature of docusaurus?

File: scripts/inject-dict.json [hunk 4]
```diff
@@ -124,5 +124,11 @@
   {
     "tpl": "polkadot_max_nominations",
     "default": "16"
+  },
+  {
+    "tpl": "vote_locking_period",
+    "default": { "kusama": 115200, "polkadot": 403200 },
+    "path": "consts.democracy.voteLockingPeriod",
+    "filters": ["blocksToDays"]
```
Comment (alfarok) [inline scripts/inject-dict.json:160]: Ah okay this is good to know.  I also see from your previous commit that the table is still only rendering Polkadot values but using conditional rendering with a table doesn't seem to work.  I was trying to see if I could find another existing example in the repo of this.

Comment (profradha) [reply scripts/inject-dict.json:160]: Yeah. Conditional rendering does not work in the tables with the way we have written the code https://github.com/w3f/polkadot-wiki/blob/70c12a9a313727e9011ef0a232a56e898cb20260/scripts/inject.mjs#L141 as the markdown syntax for table rendering may be messing with it. 

Did not want to waste much time on this. Instead, I created variables and they are usually rendered based on the locale. This was easier. Explore the code I just pushed.

**Review by DrW3RK — APPROVED** [review]
Looks good to me!

**Comment by DrW3RK** [comment]
It is a recommended practice to use variables, instead of hard coded values in the Wiki. It is an efficient way to change them everywhere at once.

**Pull request merged**